Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectQuickRun.py

```diff
@@ -220,58 +205,204 @@ def _setup(self):
+
+        # Find the units of the x axis
+        x_axis_is_temperature = len(input_workspace_names) == len(self._temperatures)
+
+        # Create the width workspace
+        width_name = self._output_ws + '_Width1'
+        self._create_width_workspace(result_workspaces, self._temperatures, run_numbers, width_name, x_axis_is_temperature)
+
+        # Create the diffusion workspace
+        diffusion_name = self._output_ws + '_Diffusion'
+        self._create_diffusion_workspace(mtd[width_name], self._temperatures, run_numbers, diffusion_name, x_axis_is_temperature)
+
+        # Group the width fit workspaces
+        group_workspaces(result_workspaces + chi_workspaces + [width_name] + [diffusion_name], self._output_ws + '_Width_Fit')
+
+    def _perform_two_peak_fits(self, workspace_names, x_min, x_max):
+        result_workspaces = []
+        chi_workspaces = []
+        run_numbers = []
+        for workspace_name in workspace_names:
+            number_of_histograms = mtd[workspace_name].getNumberHistograms()
+
+            progress_tracker = Progress(self, 0.3, 1.0, number_of_histograms+1)
+            progress_tracker.report('Finding temperature for {0}...'.format(workspace_name))
+
+            if self._temperatures is None:
+                run_number = str(mtd[workspace_name].getRunNumber())
+                run_numbers.append(run_number)
+
+            result = workspace_name[:-3] + 'fit'
+
+            for index in range(number_of_histograms):
+                progress_tracker.report('Fitting {0}-sp{1}...'.format(workspace_name, index))
+                IndirectTwoPeakFit(SampleWorkspace=workspace_name,
+                                   EnergyMin=x_min,
+                                   EnergyMax=x_max,
+                                   OutputName=result)
+
+            result_workspaces.append(result + '_Result')
+            chi_workspaces.append(result + '_ChiSq')
+        return result_workspaces, chi_workspaces, run_numbers
+
+    def _create_width_workspace(self, result_workspaces, temperatures, run_numbers, output_name, x_axis_is_temperature):
+        self._extract(result_workspaces[0], 0, output_name)
+        for index, workspace in enumerate(result_workspaces[1:]):
+            self._extract(workspace, 0, '__spectrum')
+            self._append(output_name, '__spectrum', output_name)
+
+        DeleteWorkspace(Workspace='__spectrum')
+        self._format_width_workspace(mtd[output_name], temperatures, run_numbers, x_axis_is_temperature)
+
+    @staticmethod
+    def _format_width_workspace(width_workspace, temperatures, run_numbers, x_axis_is_temperature):
+        number_of_temperatures = len(temperatures)
+        axis = NumericAxis.create(number_of_temperatures)
+        for index in range(number_of_temperatures):
+            value = float(temperatures[index]) if x_axis_is_temperature else float(run_numbers[index][-3:])
```

Comment:
I assume this -3 is to trim off the leading zeros for the run. I think there is a better way to do this. Otherwise once the run numbers get to the next order of magnitude it will stop working or we would have to always have 3 leading zeros.

Reply:
@AnthonyLim23 No. This is to make the plot versus number less cluttered/messy when using 5 or more digits. So the last 3 digits is sufficient to see a trend.

Reply:
ok, that was not clear. @Mantid-Matthew could we just have a comment explaining it?